pkg/proposal/mcms/executor.go

```diff
@@ -48,17 +47,28 @@ func NewProposalExecutor(proposal *MCMSProposal, sim bool) (*Executor, error) {
 	}, err
 }
 
+// SigningHash is used in non-ledger contexts, where
+// the EIP191 prefix is already applied, and it can be
+// signed directly with raw private key.
 func (e *Executor) SigningHash() (common.Hash, error) {
-	// Convert validUntil to [32]byte
-	var validUntilBytes [32]byte
-	binary.BigEndian.PutUint32(validUntilBytes[28:], e.Proposal.ValidUntil) // Place the uint32 in the last 4 bytes
-
-	hashToSign := crypto.Keccak256Hash(e.Tree.Root.Bytes(), validUntilBytes[:])
-	return toEthSignedMessageHash(hashToSign), nil
+	m, err := e.SigningMessage()
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return toEthSignedMessageHash(m), nil
 }
 
-func (e *Executor) SigningMessage() ([]byte, error) {
-	return ABIEncode(`[{"type":"bytes32"},{"type":"uint32"}]`, e.Tree.Root, e.Proposal.ValidUntil)
+// SigningMessage is used for ledger contexts where the ledger itself
+// will apply the EIP191 prefix.
+// Corresponds to the input here
+// https://github.com/smartcontractkit/ccip-owner-contracts/blob/main/src/ManyChainMultiSig.sol#L202
+func (e *Executor) SigningMessage() ([32]byte, error) {
+	msg, err := ABIEncode(`[{"type":"bytes32"},{"type":"uint32"}]`, e.Tree.Root, e.Proposal.ValidUntil)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	hash := crypto.Keccak256Hash(msg)
```

Comment:
the fix is here for ledger - this extra hash of the abi.encode(root, validutil) was missing